src/components/Config.tsx

```diff
@@ -0,0 +1,158 @@
+import ReactMarkdown from 'react-markdown';
+import style from '../css/markdown-styles.module.css';
+import yaml from 'js-yaml';
+
+const INDENT_SIZE = 30;
+
+const parseNameToHash = (name: string) => {
+    return name.replace(/-/g, "_");
+}
+
+interface YamlNode {
+    default: string | number | boolean;
+    description: string;
+}
+
+interface YamlData {
+    [key: string]: YamlNode | YamlData;
+}
+
+const scrollIntoView = (id: string) => {
+    const yOffset = -60;
+    const element = document.getElementById(id);
+    const y = element.getBoundingClientRect().top + window.pageYOffset + yOffset;
+
+    window.scrollTo({ top: y, behavior: 'smooth' });
+};
+
+const YamlNodeWithDescription: React.FC<{
+    name: string;
+    parentKey: string;
+    node: YamlNode;
+}> = ({ name, node, parentKey }) => {
+    const [showDescription, setShowDescription] = useState(false);
+    const [copied, setCopied] = useState(false);
+
+    node.default = node.default || 'N/A';
+    node.description = node.description || 'N/A';
+
+    const toggleDescription = () => {
+        setShowDescription(!showDescription);
+    };
+
+    useEffect(() => {
+        const hash = parentKey + "_" + parseNameToHash(name);
+        if (window.location.hash === `#${hash}`) {
+            showAndScrollIntoView(hash);
+        }
+    }, [name]);
+
+    const handleHashLinkClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+        event.preventDefault(); // Prevent the default anchor tag behavior. This causes some UI issues with scrolling.
+
+        showAndScrollIntoView(parentKey + "_" + parseNameToHash(name));
+        history.pushState(null, null, event.currentTarget.hash);
+
+        const fullURL = window.location.href.split("#")[0];
+        const hash = parentKey + "_" + name.replace(/-/g, "_");
+        navigator.clipboard.writeText(fullURL + "#" + hash);
+        setCopied(true);
+        scrollIntoView(hash);
+
+        // Hide the message after 2 seconds
+        setTimeout(() => {
+            setCopied(false);
+        }, 2000);
+    };
+
+    const showAndScrollIntoView = hash => {
+        setShowDescription(true);
+        scrollIntoView(hash);
+    }
+
+    return (
+        <div
+            style={{ paddingLeft: `${INDENT_SIZE}px` }}
+            id={parentKey + "_" + parseNameToHash(name)}
+        >
+            {showDescription ? (
+                <>
+                    <div className={'with-value-active description'} style={{ marginBottom: 10 }}>
+                        <a onClick={toggleDescription}>{name}: {node.default.toString()}</a>
+                        <a
+                            className={`config-anchor with-value-active-colour hash-link`}
+                            href={`#${parentKey + "_" + parseNameToHash(name)}`}
+                            onClick={handleHashLinkClick}
+                        ></a>
+                        {copied && <div className="copied-message">URL Copied!</div>}
```

Comment:
When this is shown, it changes the height of various things. I think it should be floating so as not not move the description block or other settings down.